pkg/git/Util.go

```diff
@@ -124,3 +125,13 @@ func CreateOrUpdateSshPrivateKeyOnDisk(gitProviderId int, sshPrivateKeyContent s
 
 	return nil
 }
+
+func IsSpaceAvailableOnDisk() bool {
+	var statFs unix.Statfs_t
+	err := unix.Statfs(GIT_BASE_DIR, &statFs)
+	if err != nil {
+		return false
+	}
+	availableSpace := int64(statFs.Bavail) * int64(statFs.Bsize)
+	return availableSpace > 100*1024*1024
```

Comment:
can introduce constant for this minimum limit. also, would be better if we derive the value from env var because at some installations this limit may not be sufficient